statement_test.go

```diff
@@ -651,3 +651,34 @@ func TestPrepareComplexQueryParameter(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, [][]any{}, nestedEmptySliceRes.Get())
 }
+
+// Define a custom type that shadows the string type
+type String256 string
+
+// Test that a custom type pointer is handled correctly by the DefaultParameterConverter
+func TestInsertCustomTypePtr(t *testing.T) {
+	db := openDbWrapper(t, ``)
+	defer closeDbWrapper(t, db)
+
+	// Create a table with a VARCHAR column
+	createTable(t, db, `CREATE TABLE test_custom_type_ptr(id INTEGER, name VARCHAR)`)
+
+	// Create a value of our custom type
+	expected := "test string"
+	customValue := String256(expected)
```

Comment:
Don't we expect the custom type, here, too, instead of falling back to `string`?
I.e., 
```Go
expected := String256("test string")
```
and
```Go
var actual String256
```